DFC.Api.JobProfiles.Repository.CosmosDb/CosmosRepository.cs

```diff
@@ -70,6 +71,31 @@ public async Task<IEnumerable<SummaryDataModel>> GetSummaryListAsync()
             return dataModels.ToArray();
         }
 
+        public async Task<IList<SegmentDetailModel>> GetSegmentData(string profileName)
+        {
+            var feedOptions = new FeedOptions
+            {
+                EnableCrossPartitionQuery = true,
+                MaxDegreeOfParallelism = -1,
+            };
+
+            var query = new SqlQuerySpec(
+                "select c.Segments from c where c.CanonicalName =  @profileName",
```

Comment:
Why are we using SQL query? Why are we not using the same code as segment apps for accessing cosmos database?

Reply:
Made repository generic following discussion